pkg/frontend/tf/config/file/file.go

```diff
@@ -163,16 +186,28 @@ func (p *FileProvider) loadJsonBytes(jsonBytes []byte) error {
 			Steps:      steps,
 		}
 
-		p.register(config)
+		p.register(registeredConfig{config: config, modifierClasses: sets.New[string]()})
 	}
 
 	for _, modifier := range modifiers {
-		var newEntries []*tfconfig.Config
+		var newEntries []registeredConfig
 
 		for _, config := range p.configs {
-			newConfig := config.Clone()
-			modifier(newConfig)
-			newEntries = append(newEntries, newConfig)
+			newConfig := config.config.Clone()
+			modifier.fn(newConfig)
+
+			if config.modifierClasses.Has(modifier.class) {
+				// incompatible combination
+				continue
```

Comment:
should modify old config for the same class?  concatenate display name or append extension

Reply:
No need, the old config is included in another iteration of this loop

Reply:
i don't understand,  whatever iteration when config1[modifier1] changed to config1[modifier1][modifier2], it modified a clone new config, will be skipped in this line, no changes will be saved